R/make-requirements.R

```diff
@@ -0,0 +1,108 @@
+
+#' Build the Requirements Specification and write it to output file(s)
+#' @importFrom purrr map walk
+#' @importFrom dplyr distinct filter group_rows slice
+#' @importFrom glue glue
+#' @importFrom knitr kable
+#' @importFrom rmarkdown render
+#' @importFrom fs dir_exists dir_create
+#' @importFrom rlang .data
+#' @param df Tibble containing stories, requirements, and tests. Created in
+#'   [create_validation_docs()].
+#' @param product_name The product you are validating, to be included in the output document.
+#' @param version The version number of the product you are validating, to be included in the output document.
+#' @param roles A data frame of user roles. If specified, this will be
+#'   inserted as a table under a "User Roles" section.
+#' @param style_dir Directory to check for a docx style reference that has the
+#'   same base name as `out_file`.
+#' @param out_file filename to write markdown file out to. Any extension will be ignored and replaced with .md
+#' @param output_dir Directory to write the output documents to. Defaults to working directory.
+#' @param word_document Logical scaler indicating whether to render a docx document
+#' @keywords internal
+make_requirements <- function(
+  df,
+  product_name,
+  version,
+  roles = NULL,
+  style_dir = NULL,
+  out_file = REQ_FILE,
+  output_dir = getwd(),
+  type = "package",
+  word_document = TRUE
+) {
+
+  template <- get_template("requirements_specification", type = type)
+
+  if (!fs::dir_exists(output_dir)) fs::dir_create(output_dir)
+  out_file <- file.path(output_dir, paste0(tools::file_path_sans_ext(out_file), ".Rmd"))
+  fs::file_copy(template, out_file)
+
+  df_story <- df %>%
+    filter(!is.na(.data$StoryId)) %>%
+    unnest(.data$tests) %>%
+    filter(!is.na(.data$passed)) %>%
+    distinct(.data$StoryId, .data$TestId, .keep_all = TRUE)
+
+  tests_by_story <- df_story %>%
+    group_by(.data$StoryId) %>%
+    group_rows() %>%
+    map(~ df_story[.x, ])
+
+  # write markdown file
+  spec_chunks <- map(tests_by_story, format_spec)
```

Comment:
I think we should change how we list the requirements for each story under a "Summary" heading. I don't know where that came from, but with all the talk of "requirements" in other docs, it seems obvious to me that we want to point out this this is (the only place) where we actually list those requirements.


<img width="667" alt="Screen Shot 2022-08-03 at 1 47 29 PM" src="https://user-images.githubusercontent.com/7806083/182675193-348f02ff-7acf-488f-af30-f83294183578.png">

I took me awhile to find where this even happens, but it's in this `format_spec()` function [here](https://github.com/metrumresearchgroup/mrgvalidate/blob/main/R/utils.R#L34). Just change `"**Summary**\n"` to `"**Requirements**\n"`.